php/class-wp-cli.php

```diff
@@ -210,30 +210,36 @@ public function assign_coauthors( $args, $assoc_args ) {
 	 * @since 3.0
 	 *
 	 * @subcommand assign-user-to-coauthor
-	 * @synopsis --user_login=<user-login> --coauthor=<coauthor>
+	 * @synopsis --user_login=<user-login> --user_id=<user-id> --coauthor=<coauthor>
 	 */
 	public function assign_user_to_coauthor( $args, $assoc_args ) {
 		global $coauthors_plus, $wpdb;
 
 		$defaults = array(
-				'user_login'        => '',
+				'user_login'        => NULL,
+				'user_id'           => NULL,
 				'coauthor'          => '',
 			);
 		$assoc_args = wp_parse_args( $assoc_args, $defaults );
 
-		$user = get_user_by( 'login', $assoc_args['user_login'] );
+		$user_id = $assoc_args['user_id'];
+		if ( $assoc_args['user_login'] ) {
+			$user = get_user_by( 'login', $assoc_args['user_login'] );
+			$user_id = $user->ID;
+		}
+
 		$coauthor = $coauthors_plus->get_coauthor_by( 'login', $assoc_args['coauthor'] );
 
-		if ( ! $user ) {
-			WP_CLI::error( __( 'Please specify a valid user_login', 'co-authors-plus' ) );
+		if ( ! $user_id ) {
```

Comment:
It might be worth checking if the `$user_id` value is an int at this point, and throwing a relevant WP_CLI error if not. Otherwise, the user could pass a non-int value in `--user_id`, and the `$wpdb->prepare()` call in l.242 could throw a non-user-friendly error when it tries to treat it as an int.

Reply:
Actually, perhaps just checking that the specified user_id corresponds to an existing user would work?

Reply:
@RobjS I've added a suggested change to coerce it earlier on, which should fix this.

Reply:
Can't check that it's an existing user, as this is specifically for use in the case where the user with that ID doesn't exist any more in the database.